nftables: update to 1.1.5

davidandreoletti opened the pull request:
Fixes https://github.com/openwrt/openwrt/issues/19482
nftables v1.1.3 introduced a regression now fixed in v1.1.4: https://github.com/openwrt/firewall4/pull/68

davidandreoletti committed:
nftables: update to 1.1.5

v1.1.3 introduced a regression now fixed in v1.1.4:
https://github.com/openwrt/firewall4/pull/68

v1.1.5 fixed a memory leak

Changelog:
https://netfilter.org/projects/nftables/files/changes-nftables-1.1.4.txt
https://netfilter.org/projects/nftables/files/changes-nftables-1.1.5.txt

Signed-off-by: David Andreoletti <david.andreoletti.dev@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20055
Signed-off-by: Ted Hess <thess@kitschensync.net>

davidandreoletti commented:
If you don't mind, I'd rather make the smallest change necessary to fix the issue reported and get it merged.

Feel free to bump it up if you need new features/bug fixes.

rockdrilla commented:
In order to update nftables to 1.1.5, you should update libnftnl to 1.3.0 first.

davidandreoletti committed:
libnftnl: update to 1.3.0

Changelog: https://git.netfilter.org/libnftnl/log/?h=libnftnl-1.3.0

Signed-off-by: David Andreoletti <david.andreoletti.dev@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20055
Signed-off-by: Ted Hess <thess@kitschensync.net>

davidandreoletti commented:
> In order to update nftables to 1.1.5, you should update libnftnl to 1.3.0 first (in separate MR, of course).

@rockdrilla 
Previous PR updated libnftnl in the same PR as nftables: https://github.com/openwrt/openwrt/pull/19288/files. I'll do the same here as nftables requires it.

davidandreoletti commented:
This job https://github.com/openwrt/openwrt/actions/runs/17790141753/job/50571513135 failed due to a bad build system package signature, before the nftables package or its dependencies were built.

Can someone with permission to restart this job, try again? Assuming it was some sort of transient error.

robimarko commented:
I restarted that job, but you have a formal issue being reported as well, please look into that

rockdrilla commented:
@davidandreoletti why not bump nftables to 1.1.5 right now?

davidandreoletti commented:
> @davidandreoletti why not bump nftables to 1.1.5 right now?

See https://github.com/openwrt/openwrt/pull/20055#issuecomment-3290958118

thess commented:
FWIW - I would prefer we update to 1.1.5 which has a couple of other good fixes (including 1 memory leak). If no substantive objections, I have already staged an update to `libnftnl 1.3.0` and `nftables 1.1.5` in my staging tree `nftables-update` and will push it tomorrow.

N.B.:  I have been testing the patch fixing time parsing in 1.1.3 for months now.

davidandreoletti commented:
> has a couple of other good fixes (including 1 memory leak)

@thess I updated the PR with v1.1.5
